### ISSUE TITLE
Slack emoji break Gutenberg/P2tenberg pasting text over from Slack

### ISSUE THREAD

**Issue by kurtextrem** [WordPress/gutenberg#41464]
### Description

I tried copying over a text from Slack to the P2tenberg editor, and for some reason nothing happened. The culprit seems to be the emoji in the HTML that is copied from Slack.

The following things are logged to the DevTools console:

<details>
<summary>Full HTML ("Received HTML")</summary>

```html
<ul data-stringify-type="unordered-list" class="p-rich_text_list p-rich_text_list__bullet" data-indent="1" data-border="0" style="box-sizing: inherit; margin: 0px; padding: 0px; list-style-type: none; color: rgb(29, 28, 29); font-family: Slack-Lato, Slack-Fractions, appleLogo, sans-serif; font-size: 15px; font-style: normal; font-variant-ligatures: common-ligatures; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(248, 248, 248); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li data-stringify-indent="1" data-stringify-border="0" style="box-sizing: inherit; margin-bottom: 0px; list-style-type: none; margin-left: 56px;">So I went out and got creative:</li><li data-stringify-indent="1" data-stringify-border="0" style="box-sizing: inherit; margin-bottom: 0px; list-style-type: none; margin-left: 56px;">I have an old Macbook (where I don't want to develop, because it is<span> </span><i data-stringify-type="italic" style="box-sizing: inherit;">slow</i>), but the Macbook now works as host for the code and the docker env</li><li data-stringify-indent="1" data-stringify-border="0" style="box-sizing: inherit; margin-bottom: 0px; list-style-type: none; margin-left: 56px;">I use VS Code Live Share so I can develop &amp; debug on my Windows machine - works good enough to be productive<span> </span><span class="c-emoji c-emoji__medium c-emoji--inline" data-qa="emoji" delay="300" data-sk="tooltip_parent" style="box-sizing: inherit; align-items: center; display: inline-flex; object-fit: contain; overflow: visible; vertical-align: top; height: auto; width: 22px; position: relative;"><img src="https://a.slack-edge.com/production-standard-emoji-assets/13.0/google-medium/1f604@2x.png" aria-label="lächeln-Emoji" alt=":lächeln:" data-stringify-type="emoji" data-stringify-emoji=":smile:" style="box-sizing: inherit; object-fit: contain; overflow: hidden; position: absolute; top: 11px; height: 22px; margin-top: -11px; width: 22px;"></span></li></ul>
```
</details>

<details>
<summary>Received plain text:</summary>

```
So I went out and got creative:
I have an old Macbook (where I don't want to develop, because it is slow), but the Macbook now works as host for the code and the docker env
I use VS Code Live Share so I can develop & debug on my Windows machine - works good enough to be productive :lächeln:
```
</details>

<details>
<summary>"Processed HTML piece" output of "Received HTML":</summary>

```html
<ul><li>So I went out and got creative:</li><li>I have an old Macbook (where I don't want to develop, because it is&nbsp;<em>slow</em>), but the Macbook now works as host for the code and the docker env</li><li>I use VS Code Live Share so I can develop &amp; debug on my Windows machine - works good enough to be productive&nbsp;<img src="https://a.slack-edge.com/production-standard-emoji-assets/13.0/google-medium/1f604@2x.png" alt=":lächeln:"></li></ul>
```
</details>

If you remove the emoji `<img>` tag, pasting works again:
<details>
<summary>HTML without the emoji</summary>

```html
<ul data-stringify-type="unordered-list" class="p-rich_text_list p-rich_text_list__bullet" data-indent="1" data-border="0" style="box-sizing: inherit; margin: 0px; padding: 0px; list-style-type: none; color: rgb(29, 28, 29); font-family: Slack-Lato, Slack-Fractions, appleLogo, sans-serif; font-size: 15px; font-style: normal; font-variant-ligatures: common-ligatures; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(248, 248, 248); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li data-stringify-indent="1" data-stringify-border="0" style="box-sizing: inherit; margin-bottom: 0px; list-style-type: none; margin-left: 56px;">So I went out and got creative:</li><li data-stringify-indent="1" data-stringify-border="0" style="box-sizing: inherit; margin-bottom: 0px; list-style-type: none; margin-left: 56px;">I have an old Macbook (where I don't want to develop, because it is<span> </span><i data-stringify-type="italic" style="box-sizing: inherit;">slow</i>), but the Macbook now works as host for the code and the docker env</li><li data-stringify-indent="1" data-stringify-border="0" style="box-sizing: inherit; margin-bottom: 0px; list-style-type: none; margin-left: 56px;">I use VS Code Live Share so I can develop &amp; debug on my Windows machine - works good enough to be productive<span> </span><span class="c-emoji c-emoji__medium c-emoji--inline" data-qa="emoji" delay="300" data-sk="tooltip_parent" style="box-sizing: inherit; align-items: center; display: inline-flex; object-fit: contain; overflow: visible; vertical-align: top; height: auto; width: 22px; position: relative;"></span></li></ul>
```
</details>

### Step-by-step reproduction instructions

1. Copy the HTML in the spoiler
2. Paste into P2tenberg

### Environment info

- P2 WP platform
- Opera GX 86.0.4363.70 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36 OPR/86.0.4363.70)
- Desktop Windows 11

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by johngodley** [comment]
Does the same thing happen in core Gutenberg? If so then it's probably best to edit this to remove mention of P2 as that is not part of the Gutenberg project. If it is specific to P2 then please report it directly to P2.

**Comment by Mamaduka** [comment]
I couldn't reproduce the issue running the latest Gutenberg plugin. Maybe the issue was fixed by #43114.

I'm going to close this, but feel free to re-open it.
